client/components/search/index.jsx

```diff
@@ -313,22 +322,32 @@ const Search = React.createClass( {
 					autoCapitalize="none"
 					dir={ this.props.dir }
 					{ ...autocorrect } />
-				{ ( searchValue || this.state.isOpen ) ? this.closeButton() : null }
+					{ this.closeButton() }
 			</div>
 		);
 	},
 
 	closeButton: function() {
-		return (
-			<span
-				onTouchTap={ this.closeSearch }
-				tabIndex="0"
-				onKeyDown={ this.closeListener }
-				aria-controls={ 'search-component-' + this.state.instanceId }
-				aria-label={ i18n.translate( 'Close Search', { context: 'button label' } ) }>
-			<Gridicon icon="cross" className={ 'search-close__icon' + ( this.props.dir ? ' ' + this.props.dir : '' ) } />
-			</span>
-		);
+		const gridIconClass = classNames ( {
+			'search-close__icon': true,
+			rtl: this.props.dir === 'rtl',
+			ltr: this.props.dir === 'ltr',
+		} );
```

Comment:
And given `'search-close__icon': true`, actually

```js
const gridIconClass = classNames ( 'search-close__icon', this.props.dir );
```

Maybe in combination with changing `dir`'s `propType` to 

```js
PropTypes.oneOf( ['ltr', 'rtl'] )
```